### PR TITLE
DO NOT SQUASH: Issue 2480 merge automation

### PR DESCRIPTION
**Pull request by prb112**
This updates the issue 2480.

@lmsurpre if you can review this part... 
PR https://github.com/IBM/FHIR/pull/2762/commits/d604d3a448d3e863ceb44d6c66fcae51523e3bc7 <-- this

### CODE REVIEW

**Review by csandersdev — APPROVED** [review]
Per Slack discussion

File: fhir-parent/pom.xml
Comment (lmsurpre) [inline fhir-parent/pom.xml:282]: will this prevent us from controlling the version of jackson-databind used by COS?

File: fhir-parent/java.header
Comment (lmsurpre) [inline fhir-parent/java.header:2]: I'd rather keep this part loose.  We require DCO signoff for contributions but not copyright assignment.
Maybe just go with `.+` instead of expecting `IBM Corp.`?

The character group at the end looks a bit odd to me as well, should it be `( )` instead of `[ ]`?
```suggestion
^ \* \(C\) Copyright .+ \d\d\d\d(, \d\d\d\d)*
```

File: fhir-parent/checkstyle.xml
Comment (lmsurpre) [inline fhir-parent/checkstyle.xml:9]: we can finally stop manually checking headers!

File: docs/src/pages/guides/FHIRClinicalReasoningGuide.md
Comment (lmsurpre) [inline docs/src/pages/guides/FHIRClinicalReasoningGuide.md:13]:
```suggestion
The IBM FHIR Server's module builds upon the work of other open source projects including [cqframework/clinical_quality_language](https://github.com/cqframework/clinical_quality_language/), [DBCG/cql_engine](https://github.com/DBCG/cql_engine/), and [DBCG/cql-evaluator](https://github.com/DBCG/cql-evaluator/) projects and much of the work is to expose these functions with the necessary hooks to perform CQL to ELM translation and ELM evaluation natively inside the IBM FHIR server or externally using an IBM FHIR Server client and model objects. Additionally, support is provided for evaluating and reporting on clinical quality measures and care gaps.
```

File: fhir-parent/pom.xml
Comment (prb112) [reply fhir-parent/pom.xml:282]: great catch... I'll fix this

Comment (prb112) [reply fhir-parent/pom.xml:282]: fixed, waiting on the CI

File: cql/fhir-cql/README.md
Comment (lmsurpre) [inline cql/fhir-cql/README.md:3]:
```suggestion
This module implements an IBM FHIR Server backend for the [third-party open source CQL engine](https://github.com/DBCG/cql_engine/) including model resolution and type conversion for CQL to FHIR data operations, base classes for building IBM FHIR Server retrieve providers, as well as, CQL translation support. 
```

File: cql/fhir-cql/src/main/java/com/ibm/fhir/cql/engine/model/FHIRModelResolver.java
Comment (lmsurpre) [inline cql/fhir-cql/src/main/java/com/ibm/fhir/cql/engine/model/FHIRModelResolver.java:492]: just noting the TODO for future considerations

File: fhir-parent/pom.xml
Comment (lmsurpre) [reply fhir-parent/pom.xml:282]: I see you added jackson-databind back, but I'm thiinking maybe it should be using this new `${jackson.version}` variable.  Thoughts?
BTW, this is such a big PR that I'm totally fine to merge as-is and tweak the pom dependencies in a standalone one.

File: fhir-server-test/src/test/resources/testdata/Library-CompileFailure.json
Comment (lmsurpre) [inline fhir-server-test/src/test/resources/testdata/Library-CompileFailure.json:19]: i confirmed the base64 entry is non-malicioous:
```
library "InlineLibrary" version '1.0.0'
using "FHIR" version '4.0.1'
include "FHIRHelpers" version '4.0.1'
context Patient
define InlineExpression:
	[NotAResource]
```

File: fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathBaseVisitor.java
Comment (lmsurpre) [inline fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathBaseVisitor.java:7]: this is a generated file, so i'm not sure if the header is needed or not.  i wonder if there's a way to make antlr generate it...

File: fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathParser.java
Comment (lmsurpre) [inline fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathParser.java:7]: this is a generated file, so i'm not sure if the header is needed or not.  i wonder if there's a way to make antlr generate it...

File: fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathVisitor.java
Comment (prb112) [reply fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathVisitor.java:7]: same comment as Pathparser

**Review by lmsurpre — APPROVED** [review]
I can't attest to the functionality of the code, but I think the overall structure meets the bar for getting this in as an exciting new experimental feature.

File: fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathBaseVisitor.java
Comment (prb112) [reply fhir-path/src/main/java/com/ibm/fhir/path/FHIRPathBaseVisitor.java:7]: same comment as it is generated

File: fhir-parent/pom.xml
Comment (prb112) [reply fhir-parent/pom.xml:282]: so jackson gets kind of tricky as they version independently, leaving as is is OK, and case-by-case basis we can change it

Comment (prb112) [reply fhir-parent/pom.xml:282]: aka - I think that is a standalone activity

File: cql/operation/fhir-operation-cpg/src/test/resources/SupplementalDataElements-2.0.0.json
Comment (lmsurpre) [inline cql/operation/fhir-operation-cpg/src/test/resources/SupplementalDataElements-2.0.0.json:101]:
```
library SupplementalDataElements version '2.0.0'

using FHIR version '4.0.1'

include FHIRHelpers version '4.0.1' called FHIRHelpers

valueset "ONC Administrative Sex": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1'
valueset "Race": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.114222.4.11.836'
valueset "Ethnicity": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.114222.4.11.837'
valueset "Payer": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.114222.4.11.3591'

context Patient

define "SDE Ethnicity":
  (flatten (
    Patient.extension Extension
      where Extension.url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity'
        return Extension.extension
  )) E
    where E.url = 'ombCategory'
      or E.url = 'detailed'
    return E.value as Coding

define "SDE Payer":
    [Coverage: type in "Payer"] Payer
      return {
        code: Payer.type,
        period: Payer.period
      }

define "SDE Race":
  (flatten (
    Patient.extension Extension
      where Extension.url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-race'
        return Extension.extension
  )) E
    where E.url = 'ombCategory'
      or E.url = 'detailed'
    return E.value as Coding

define "SDE Sex":
  case
    when Patient.gender = 'male' then Code { code: 'M', system: 'http://hl7.org/fhir/v3/AdministrativeGender', display: 'Male' }
    when Patient.gender = 'female' then Code { code: 'F', system: 'http://hl7.org/fhir/v3/AdministrativeGender', display: 'Female' }
    else null
  end
```

**Pull request merged**
